prime-router/src/main/kotlin/azure/ReportFunction.kt

```diff
@@ -191,7 +200,7 @@ class ReportFunction : Logging {
                     )
 
                     val receivedBlobInfo = workflowEngine.recordReceivedReport(
-                        report, validatedRequest.content.toByteArray(), sender, actionHistory, payloadName
+                        report, rawBody, sender, actionHistory, payloadName
```

Comment:
You have already calculated the digest - maybe you should also pass it down to the blob store, rather than calculating it again there?

Reply:
Calculating the digest is cheap, and the way we are passing the byte array around (from numerous place) makes it more difficult to pass this in. As this gets refactored for SRP more accurately this will be possible. Adding a comment to this effect.